backend/src/main/java/com/programmers/pcquotation/global/rq/Rq.kt

```diff
@@ -0,0 +1,85 @@
+package com.programmers.pcquotation.global.rq
+
+import org.springframework.http.ResponseCookie
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.stereotype.Component
+import org.springframework.web.context.annotation.RequestScope
+import com.programmers.pcquotation.domain.estimaterequest.exception.NullEntityException
+import com.programmers.pcquotation.domain.member.entitiy.Member
+import com.programmers.pcquotation.global.enums.UserType
+import com.programmers.pcquotation.global.security.CustomUserDetails
+import com.programmers.pcquotation.global.security.CustomUserDetailsService
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+
+import org.springframework.beans.factory.annotation.Autowired
+
+@RequestScope
+@Component
+class Rq @Autowired constructor(
+    private val req: HttpServletRequest,
+    private val resp: HttpServletResponse,
+    private val customUserDetailsService: CustomUserDetailsService
+) {
+
+    fun setLogin(member: Member) {
+        val user = CustomUserDetails(member)
+        val authentication: Authentication = UsernamePasswordAuthenticationToken(
+            user, user.password, user.authorities
+        )
+        SecurityContextHolder.getContext().authentication = authentication
+    }
+
+    fun getMember(): Member? {
+        val authorization = getHeader("Authorization") ?: return null
+        if (!authorization.startsWith("Bearer ")) throw NullEntityException()
+
+        val token = authorization.removePrefix("Bearer ")
+        val tokenBits = token.split(" ", limit = 3)
+
+        if (tokenBits.size < 3) return null
+
+        val userType = UserType.fromString(tokenBits[2])
+        return (SecurityContextHolder.getContext().authentication?.principal as? UserDetails)?.let {
+            customUserDetailsService.loadUserByUsername(it.username, userType)
+        }
+    }
```

Comment:
```kotlin
val memberFromRq: AuthRequest
        get() {
            val member = rq.getMember()!!
            val auth = member.authorities
            val userType = UserType.fromString(auth.toString()).toString()
            return AuthRequest(userType)
        }
```
member를 받을 때 !!로 받으시는 것 같습니다.
이렇게 하기 보다는, 미리 null 처리를 한 뒤 반환하는 게 더 좋습니다.
```suggestion
    fun getMember(): Member {
        val authorization = getHeader("Authorization") ?: return null
        if (!authorization.startsWith("Bearer ")) throw NullEntityException()

        val token = authorization.removePrefix("Bearer ")
        val tokenBits = token.split(" ", limit = 3)

        if (tokenBits.size < 3) // null -> 예외 

        val userType = UserType.fromString(tokenBits[2])
        vval authentication = SecurityContextHolder.getContext().authentication ?: // null -> 예외
        val userDetails = authentication.principal as UserDetails

        return customUserDetailsService.loadUserByUsername(userDetails.username, userType)
    }
```
대략 이런 느낌일 것 같습니다!